### DISCUSSION TITLE
Run code when property changes in v.5.5.3

### DISCUSSION THREAD

**Discussion by oskarjf** [MarimerLLC/csla#3278] [Questions] [answered]
What is the best way to monitor if a property changes and do some checks?
In older versions it was possible to have if statement checks within the setter.

**Comment by jonnybee** [comment, marked as answer]
What kind of checks do you want to do? 

I will usually create BusinessRules to react to a property changed.

**Reply by oskarjf** [reply]
Current functionality is checking what the value is being set and if it meets the criteria sets the value of another property.

**Reply by oskarjf** [reply]
set
{
  if (value == "a")
    PropB = 1;
  SetProperty<string>(StringProperty, value);
}

**Reply by jonnybee** [reply]
You can easily write a rule that sets other property values by context.AddOutValue(). 

See: https://github.com/MarimerLLC/csla/blob/main/Samples/RuleTutorial/AsyncLookupRule/Rules/AsyncLookupCustomer.cs

**Reply by oskarjf** [reply]
Thanks Jonny. That covers it.
